### PR TITLE
Adding sFTP connector

### PR DESCRIPTION
**Pull request by Luishfs**
# Summary
Adding sFTP tap to data integration pipelines

# Tests
Tested on a remote sFTP server to a local postgreSQL instance

Default INCREMENTAL run:
Screenshot from 2023-08-08 16-38-00

Screenshot from 2023-08-08 15-52-22

State reading:
Screenshot from 2023-08-08 15-52-33

End data:

Screenshot from 2023-08-08 16-00-03

Screenshot from 2023-08-10 20-30-28

Table updated schema:
Screenshot from 2023-08-10 20-31-32

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_integrations/mage_integrations/sources/sftp/tap_sftp/tap.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                                      (),
+                                      "table-key-properties")
+        replication_method = stream.replication_method
+        write_schema(stream_name,
+                     stream.schema.to_dict(),
+                     key_properties,
+                     replication_method=replication_method)
+
+        logger.info(f"{stream_name}: Starting sync")
+        counter_value = sync_stream(config, state, stream, sftp_client)
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/sftp/tap_sftp/tap.py:55]: Can we pass the logger to `sync_stream` and `sync_file` as well?

**Comment by wangxiaoyou1993** [comment]
Does incremental sync work correctly? I saw it uses `modified_since` as bookmark. Do we include it in the schema?

File: mage_integrations/mage_integrations/sources/sftp/tap_sftp/tap.py (hunk 1 above)
Comment (Luishfs) [reply mage_integrations/mage_integrations/sources/sftp/tap_sftp/tap.py]: Added on e690d19139e25188db4a840f2ba56ffc1575f7c0

**Comment by Luishfs** [comment]
Incremental works using the last modified time all the files were modified.
I included this information as `_sdc_source_last_modified` on the table schema, using `datetime` format

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**